app/controllers/automated_tests_controller.rb

```diff
@@ -220,4 +222,17 @@ def required_params
   def assignment_params
     params.require(:assignment).permit(*required_params)
   end
+
+  def find_or_create_test_student(assignment_id)
+    test_group = Group.find_or_create_by(group_name: 'test_student_group')
+    user = TestStudent.find_or_create_by(user_name: 'test_student',
+                                         first_name: 'Test', last_name: 'Student', hidden: true)
```

Comment:
I don't think the hidden is necessary here to create the TestStudent and in case the TestStudent has already been created but the hidden value has been switched, we'll end up accidentally creating a new one that we don't need.

Reply:
Actually, the first_name and last_name could be accidentally changed later on as well. Instead, set the first and last name in the block passed to `find_or_create_by` instead (and do the same for the membership_status when you find or create the StudentMembership below).